### PR TITLE
178 header

### PR DESCRIPTION
**Pull request by mikes-lunarg**
Header update:
- Update known-good
- Modify command_validation_generator.py to use "UNASSIGNED" ids in case
  of a mismatch between vk.xml and validusage.json
- Generate source

VUID reconciliation:
- `VUID-VkCopyDescriptorSet-srcSet-01919` replaced with:
  - `VUID-VkCopyDescriptorSet-srcSet-04885` (VK_VALVE_mutable_descriptor_type)
  - `VUID-VkCopyDescriptorSet-srcSet-04886` (core)
- `VUID-VkCopyDescriptorSet-srcSet-01921` replaced with:
  - `VUID-VkCopyDescriptorSet-srcSet-04887` (VK_VALVE_mutable_descriptor_type)
  - `VUID-VkCopyDescriptorSet-srcSet-04888` (core)
- Removed:
  - `VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03666`
  - `VUID-vkBuildAccelerationStructuresKHR-pInfos-03666`
  - `VUID-vkCmdBuildAccelerationStructuresIndirectKHR-pInfos-03666`

### CODE REVIEW

File: layers/descriptor_sets.cpp [hunk 1]
```diff
@@ -1683,19 +1683,35 @@ bool CoreChecks::ValidateCopyUpdate(const VkCopyDescriptorSet *update, const Des
...
-                  << report_data->FormatHandle(update->dstSet)
-                  << ") layout must: also have been created without the "
-                     "ename:VK_DESCRIPTOR_SET_LAYOUT_CREATE_UPDATE_AFTER_BIND_POOL_BIT flag set";
-        *error_msg = error_str.str();
-        return false;
+    if (device_extensions.vk_valve_mutable_descriptor_type) {
+        if (!(src_layout->GetCreateFlags() & (VK_DESCRIPTOR_SET_LAYOUT_CREATE_UPDATE_AFTER_BIND_POOL_BIT |
+                                              VK_DESCRIPTOR_SET_LAYOUT_CREATE_HOST_ONLY_POOL_BIT_VALVE)) &&
+            (dst_layout->GetCreateFlags() & VK_DESCRIPTOR_SET_LAYOUT_CREATE_UPDATE_AFTER_BIND_POOL_BIT)) {
+            *error_code = "VUID-VkCopyDescriptorSet-srcSet-04885";
```
Comment (ncesario-lunarg) [inline layers/descriptor_sets.cpp:1690]: For some reason I'm not seeing this in the 1.2.178 pdf ("with extensions"); though I see `VK_VALVE_mutable_descriptor_type` documented in the PDF as well as these VUIDs in Vulkan-Docs?

File: scripts/command_validation_generator.py [hunk 2]
```diff
@@ -249,8 +249,8 @@ def commandRecordingList(self):
         for name, cmdinfo in sorted(self.commands.items()):
             vuid = 'VUID-' + name + '-commandBuffer-recording'
             if vuid not in self.valid_vuids:
-                print("Error: Could not find %s in validusage.json\n", vuid)
-                sys.exit(1)
+                print("Warning: Could not find {} in validusage.json".format(vuid))
+                vuid = vuid.replace('VUID-', 'UNASSIGNED-')
```
Comment (ncesario-lunarg) [inline scripts/command_validation_generator.py:253]: Just curious what caused this as it looks like kind of a fundamental change?

**Review by ncesario-lunarg — APPROVED** [review]
LGTM! Couple questions for my own understanding.

File: scripts/command_validation_generator.py (hunk 2 above)
Comment (mikes-lunarg) [reply scripts/command_validation_generator.py:253]: There is a new extension `VK_NVX_binary_import` which only exists in the xml and spec extension appendix right now, so it doesn't get any VUIDs in the spec body or validusage.json.

I'm basically saying that vk.xml is the source of truth here, and we should generate the validation even if we can't find the string in validusage.json. Switching `VUID-` prefix to `UNASSIGNED-` is to silence the vk_validation_stats.py consistency warnings. This approach of validating anyway with a non-VUID name is similar what we do when some of the stateless checks don't line up perfectly with implicit spec VUIDs: https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/20f3d5efd2e75b19b3ac5bdb167bf53a377ac24e/scripts/parameter_validation_generator.py#L1011-L1023

**Review by jeremyg-lunarg — APPROVED** [review]

File: layers/descriptor_sets.cpp (hunk 1 above)
Comment (mikes-lunarg) [reply layers/descriptor_sets.cpp:1690]: `-04885` and `-04887` only exist in the extension spec, `-04886` and `-04888` only exist in the core spec. This is due to how the spec conditionally includes extension content (and excludes superseded core content) using `#ifdef`s

Its a lot easier to see in the spec source:
https://github.com/KhronosGroup/Vulkan-Docs/blob/dfe3bcd0e1b7239e2d6ae8b6afe563780a711854/chapters/descriptorsets.txt#L3654-L3670

File: scripts/command_validation_generator.py (hunk 2 above)
Comment (ncesario-lunarg) [reply scripts/command_validation_generator.py:253]: Thanks for the explanation!

**Pull request merged**